pyanaconda/modules/user/user.py

```diff
@@ -0,0 +1,125 @@
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.dbus import DBus
+from pyanaconda.dbus.constants import MODULE_USER_NAME, MODULE_USER_PATH
+from pyanaconda.core.signal import Signal
+from pyanaconda.modules.base import KickstartModule
+from pyanaconda.modules.user.user_interface import UserInterface
+from pyanaconda.modules.user.user_kickstart import UserKickstartSpecification
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class UserModule(KickstartModule):
+    """The User module."""
+
+    def __init__(self):
+        super().__init__()
+        self.root_password_set_changed = Signal()
+        self._root_password_set = False
+        self._root_password = ""
+        self._root_password_is_crypted = False
+
+        self.root_account_locked_changed = Signal()
+        self._root_account_locked = False
+
+    def publish(self):
+        """Publish the module."""
+        DBus.publish_object(UserInterface(self), MODULE_USER_PATH)
+        DBus.register_service(MODULE_USER_NAME)
+
+    @property
+    def kickstart_specification(self):
+        """Return the kickstart specification."""
+        return UserKickstartSpecification
+
+    def process_kickstart(self, data):
+        """Process the kickstart data."""
+        log.debug("Processing kickstart data...")
+        self.set_root_password(data.rootpw.password, crypted=data.rootpw.isCrypted)
+        self.set_root_account_locked(data.rootpw.lock)
+
+    def generate_kickstart(self):
+        """Return the kickstart string."""
+        log.debug("Generating kickstart data...")
+        data = self.get_kickstart_handler()
+        data.rootpw.password = self._root_password
+        data.rootpw.isCrypted = self._root_password_is_crypted
+        data.rootpw.lock = self.root_account_locked
+        return str(data)
+
+    @property
+    def root_password(self):
+        """The root password.
+
+        :returns: root password (might be crypted)
+        :rtype: str
+        """
+        return self._root_password
+
+    @property
+    def root_password_is_crypted(self):
+        """Is the root password crypted ?
+
+        :returns: if root password is crypted
+        :rtype: bool
+        """
+        return self._root_password_is_crypted
+
+    def set_root_password(self, root_password, crypted):
+        """Set the crypted root password.
+
+        :param str root_password: root password
+        :param bool crypted: if the root password is crypted
+        """
+        self._root_password = root_password
+        self._root_password_is_crypted = crypted
+        self.root_password_set_changed.emit()
+        log.debug("Root password set.")
+
+    def clear_root_password(self):
+        """Clear any set root password."""
+        self._root_password = ""
+        self._root_password_is_crypted = False
+        self.root_password_set_changed.emit()
```

Comment:
Please, keep the name of the signal consistent with the name of the property. In this case, it should be `root_password_is_set_changed`.

Reply:
Good point! Fixed.